jit/src/main/java/org/dcsa/conformance/standards/jit/party/JitConsumer.java

```diff
@@ -0,0 +1,133 @@
+      Map<String, ? extends Collection<String>> orchestratorAuthHeader) {
+    super(
+        apiVersion,
+        partyConfiguration,
+        counterpartConfiguration,
+        persistentMap,
+        webClient,
+        orchestratorAuthHeader);
+  }
+
+  @Override
+  protected void exportPartyJsonState(ObjectNode targetObjectNode) {
+    // No state to export.
+  }
+
+  @Override
+  protected void importPartyJsonState(ObjectNode sourceObjectNode) {
+    // No state to import.
+  }
+
+  @Override
+  protected void doReset() {
+    // No state to reset.
+  }
+
+  @Override
+  protected Map<Class<? extends ConformanceAction>, Consumer<JsonNode>> getActionPromptHandlers() {
+    return Map.ofEntries(
+        Map.entry(SupplyScenarioParametersAction.class, this::supplyScenarioParameters),
+        Map.entry(JitTimestampAction.class, this::timestampRequest));
+  }
+
+  private void supplyScenarioParameters(JsonNode actionPrompt) {
+    log.info("JitConsumer.supplyScenarioParameters({})", actionPrompt.toPrettyString());
+
+    SuppliedScenarioParameters parameters = createSuppliedScenarioParameters();
+    asyncOrchestratorPostPartyInput(
+        actionPrompt.required("actionId").asText(), parameters.toJson());
+
+    addOperatorLogEntry(
+        "Submitted SuppliedScenarioParameters: %s".formatted(parameters.toJson().toPrettyString()));
+  }
+
+  public static SuppliedScenarioParameters createSuppliedScenarioParameters() {
+    return new SuppliedScenarioParameters(
+        UUID.randomUUID().toString(), UUID.randomUUID().toString(), UUID.randomUUID().toString());
+  }
+
+  private void timestampRequest(JsonNode actionPrompt) {
+    log.info("JitConsumer.timestampRequest({})", actionPrompt.toPrettyString());
+
+    JitTimestampType timestampType =
+        JitTimestampType.valueOf(actionPrompt.required("timestampType").asText());
+    if (timestampType != JitTimestampType.REQUESTED) {
+      throw new UserFacingException(
+          "Only REQUESTED timestamps are supported for a Consumer party.");
+    }
+
+    DynamicScenarioParameters dsp = DynamicScenarioParameters.fromJson(actionPrompt.path("dsp"));
+    JitTimestamp timestamp =
+        JitProvider.getTimestampForType(timestampType, dsp.currentTimestamp());
+
+    syncCounterpartPut(
+        JitStandard.PORT_CALL_SERVICES_URL + timestamp.portCallServiceID() + "/timestamp",
+        timestamp.toJson());
+
+    addOperatorLogEntry(
+        "Submitted %s timestamp for: %s".formatted(timestampType, timestamp.dateTime()));
+  }
+
+  @Override
+  public ConformanceResponse handleRequest(ConformanceRequest request) {
+    log.info("JitConsumer.handleRequest()");
+    int statusCode = 204;
+    if (request.message().body().getJsonBody().has("specification")) {
+      String portCallServiceType =
+          request
+              .message()
+              .body()
+              .getJsonBody()
+              .get("specification")
+              .get("portCallServiceType")
+              .asText();
```

Comment:
Potential NPE with malformed requests. Use `.path` instead of `.get` to avoid it.

Reply:
(note, `.has("specification")` does not protect `get("portCallServiceType").asText` from doing an NPE)